Compiletime/syntax_check.py

```diff
@@ -71,3 +71,55 @@ def check_PrintStmt(tokens):
             return False
 
     return True
+
+def check_Expr(tokens):
+    """
+    Check whether the given expr is a valid expr or not.
+
+    Args:
+        tokens (list): A stream of tokens representing the statement.
+
+    Returns:
+        bool: True if the syntax is valid, False otherwise.
+
+    Rule to check:
+    * An expression should not contain any keyword(for now).
+    * An expression should not contain any unclosed bracket.
+    """
+    valid = True
+    reason = ''
+
+    def has_unclosed_brackets(_tokens):
+        stack = []
+
+        # Define mappings for opening and closing brackets
+        bracket_map = {
+            '(': ')',
+            '[': ']',
+            '{': '}',
+        }
+
+        for token in _tokens:
+            if token.token in bracket_map.keys():  # Use token.token as the key
+                # If it's an opening bracket, push it onto the stack
+                stack.append(token)
+            elif token.token in bracket_map.values():
```

Comment:
try to avoid nested statements you can do this 

```
if token.token in bracket_map:
            stack.append(token.token)
        elif token.token in bracket_map.values() and (not stack or bracket_map[stack.pop()] != token.token):
            return True
```